### PR TITLE
LPS-187257 Add doAsGroupId to session_click request to avoid redundant calculations

### PR DESCRIPTION
**Pull request by antonio-ortega**
Hey guys,

This is actually a forward port of [LPS-135925](https://issues.liferay.com/browse/LPS-135925) (see [this commit](https://github.com/liferay/liferay-portal-ee/commit/c462aefa8ebe434274e5fd986b1b3cfd00f80d6a)).

Thanks.

Regards.

/cc @jgarcialr

### CODE REVIEW

**Comment by antonio-ortega** [comment]
ci:test:sf

**Comment by antonio-ortega** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
@antonio-ortega is this a duplicate of https://github.com/liferay-frontend/liferay-portal/pull/3395#issuecomment-1584368486? Or will your PR need to rebase on that?

**Comment by antonio-ortega** [comment]
Hey @bryceosterhaus , I'd need to rebase on your pull request.

**Comment by bryceosterhaus** [comment]
@antonio-ortega sounds good. I just forwarded to Brian, so when he merges you can rebase

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/fetch.es.js [hunk 1]
```diff
@@ -67,6 +67,20 @@ export default function defaultFetch(resource, init = {}) {
...
+		if (doAsGroupId) {
+			resourceURL.searchParams.set('doAsGroupId', doAsGroupId);
+			resourceLocation = resourceURL.toString();
+
+			if (typeof resource === 'string') {
+				resource = resourceLocation;
+			}
+			else {
+				resource = {...resource, url: resourceLocation};
+			}
```
Comment (dsanz) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/fetch.es.js:82]: IS there any reason why we need to repeat the block?:

```
			resourceLocation = resourceURL.toString();

			if (typeof resource === 'string') {
				resource = resourceLocation;
			}
			else {
				resource = {...resource, url: resourceLocation};
			}
```

It's 3 times there. Looks like its primary goal is to  update `resourceLocation` and `resourceURL` but the part concerning `resource` could be moved to be run right before the actual `fetch` call right? cc/ @bryceosterhaus  perhaps I'm not seeing something here, let me know

Comment (antonio-ortega) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/fetch.es.js:82]: Yes, actually is what we do in liferay-frontend#3395. I'll just rebase this on top of that one.

Comment (dsanz) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/fetch.es.js:82]: great!

**Comment by mateusralv** [comment]
Hi @antonio-ortega I found some failures that seem related to your PR, could you take a look?

`js-unit-jdk8/0/0`
```
 [exec]     console.error ../../../node_modules/warning/warning.js:34
     [exec]       Warning: Button Accessibility: Component has only the Icon declared. Define an `aria-label` or `aria-labelledby` attribute that labels the interactive button that screen readers can read. The `title` attribute is optional but consult your design team.
     [exec] 
     [exec]   ��� AccountSelector ��� When no account is selected ��� must update the remote selected account when an account item is clicked
     [exec] 
     [exec]     No fallback response defined for POST to [object Object]
     [exec] 
     [exec]       89 | 
     [exec]       90 | 	// eslint-disable-next-line @liferay/portal/no-global-fetch
     [exec]     > 91 | 	return fetch(resource, {...config, ...init, headers});
     [exec]          | 	       ^
     [exec]       92 | }
     [exec]       93 | 
     [exec] 
     [exec]       at FetchMock.fetch [as fetchMock] (../../../node_modules/fetch-mock/src/fetch-mock.js:97:10)

...

 [exec] 
     [exec]       182 | 		it('must display the orders search autocomplete component"', () => {
     [exec]       183 | 			expect(
     [exec]     > 184 | 				renderedComponent.getByPlaceholderText(/search-order/)
     [exec]           | 				                  ^
     [exec]       185 | 			).toBeInTheDocument();
     [exec]       186 | 		});
     [exec]       187 | 
     [exec] 
     [exec]       at Object.getElementError (../../../node_modules/@testing-library/dom/dist/config.js:38:19)
     [exec]       at ../../../node_modules/@testing-library/dom/dist/query-helpers.js:90:38
     [exec]       at ../../../node_modules/@testing-library/dom/dist/query-helpers.js:62:17
     [exec]       at getByPlaceholderText (../../../node_modules/@testing-library/dom/dist/query-helpers.js:111:19)
     [exec]       at Object.<anonymous> (test/components/AccountSelector/AccountSelector.js:184:23)
...
 [exec] FAIL test/liferay/util/fetch.es.js
     [exec]   ��� Liferay.Util.fetch ��� adds auth-token and credentials if origin is the same
     [exec] 
     [exec]     expect(jest.fn()).toHaveBeenCalledWith(...expected)
     [exec] 
     [exec]     - Expected
     [exec]     + Received
     [exec] 
     [exec]     - "http://localhost/o/test",
     [exec]     + "http://localhost/o/test?doAsGroupId=20126",
     [exec]       {"credentials": "include", "headers": {Symbol(map): {"x-csrf-token": ["default-mocked-auth-token"]}}},
     [exec] 
     [exec]     Number of calls: 1
     [exec] 
     [exec]       43 | 		};
     [exec]       44 | 
     [exec]     > 45 | 		expect(fetch).toHaveBeenCalledWith(sameOriginUrl, mergedInit);
     [exec]          | 		              ^
     [exec]       46 | 	});
     [exec]       47 | 
     [exec]       48 | 	it('overrides default auth-token', () => {
     [exec] 
     [exec]       at Object.<anonymous> (test/liferay/util/fetch.es.js:45:17)
     [exec]       at processTicksAndRejections (node:internal/process/task_queues:96:5)
     [exec] 
     [exec]   ��� Liferay.Util.fetch ��� overrides default auth-token
     [exec] 
     [exec]     expect(jest.fn()).toHaveBeenCalledWith(...expected)
     [exec] 
     [exec]     - Expected
     [exec]     + Received
     [exec] 
     [exec]     - "http://localhost/o/test",
     [exec]     + "http://localhost/o/test?doAsGroupId=20126",
     [exec]       {"credentials": "include", "headers": {Symbol(map): {"x-csrf-token": ["asdf"]}}},
     [exec] 
     [exec]     Number of calls: 1
     [exec] 
     [exec]       60 | 		};
     [exec]       61 | 
     [exec]     > 62 | 		expect(fetch).toHaveBeenCalledWith(sameOriginUrl, mergedInit);
     [exec]          | 		              ^
     [exec]       63 | 	});
     [exec]       64 | 
     [exec]       65 | 	it('overrides default settings with given settings', () => {
     [exec] 
     [exec]       at Object.<anonymous> (test/liferay/util/fetch.es.js:62:17)
     [exec]       at processTicksAndRejections (node:internal/process/task_queues:96:5)
     [exec] 
     [exec]   ��� Liferay.Util.fetch ��� overrides default settings with given settings
     [exec] 
     [exec]     expect(jest.fn()).toHaveBeenCalledWith(...expected)
     [exec] 
     [exec]     - Expected
     [exec]     + Received
     [exec] 
     [exec]     - "http://localhost/o/test",
     [exec]     + "http://localhost/o/test?doAsGroupId=20126",
     [exec]       {"credentials": "omit", "headers": {Symbol(map): {"x-csrf-token": ["efgh"]}}},
     [exec] 
     [exec]     Number of calls: 1
     [exec] 
     [exec]       82 | 		};
     [exec]       83 | 
     [exec]     > 84 | 		expect(fetch).toHaveBeenCalledWith(url, mergedInit);
     [exec]          | 		              ^
     [exec]       85 | 	});
     [exec]       86 | 
     [exec]       87 | 	it('overrides default settings with given settings', () => {
     [exec] 
     [exec]       at Object.<anonymous> (test/liferay/util/fetch.es.js:84:17)
     [exec]       at processTicksAndRejections (node:internal/process/task_queues:96:5)
     [exec] 
     [exec]   ��� Liferay.Util.fetch ��� includes path context when it is defined on themeDisplay.getPathContext()
     [exec] 
     [exec]     TypeError: Liferay.ThemeDisplay.getScopeGroupId is not a function
     [exec] 
     [exec]       69 | 		}
     [exec]       70 | 
     [exec]     > 71 | 		const doAsGroupId = Liferay.ThemeDisplay.getScopeGroupId();
     [exec]          | 		                                         ^
     [exec]       72 | 
     [exec]       73 | 		if (doAsGroupId) {
     [exec]       74 | 			resourceURL.searchParams.set('doAsGroupId', doAsGroupId);
     [exec] 
     [exec]       at defaultFetch (src/main/resources/META-INF/resources/liferay/util/fetch.es.js:71:44)
     [exec]       at Object.<anonymous> (test/liferay/util/fetch.es.js:222:3)
     [exec]       at processTicksAndRejections (node:internal/process/task_queues:96:5)
     [exec] 
     [exec] 
     [exec] Test Suites: 1 failed, 57 passed, 58 total
     [exec] Tests:       4 failed, 3 skipped, 458 passed, 465 total
```

**Comment by antonio-ortega** [comment]
Thanks for catching this, @mateusralv. I'm closing this PR now since I think I need to redo it to avoid those errors.

**Pull request closed without merging**